plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -650,4 +770,101 @@ private static Optional<ColumnMapping> getUnsignedMapping(JdbcTypeHandle typeHan
+            return handle.createQuery("""
+                        SELECT max(CARDINALITY) AS row_count FROM INFORMATION_SCHEMA.STATISTICS
+                        WHERE TABLE_SCHEMA = :schema AND TABLE_NAME = :table_name
+                        """)
+                    .bind("schema", remoteTableName.getCatalogName().orElse(null))
+                    .bind("table_name", remoteTableName.getTableName())
+                    .mapTo(Long.class)
+                    .findOne()
+                    .orElse(null);
+        }
+
+        Map<String, AnalyzeColumnStatistics> getColumnStatistics(JdbcTableHandle table)
+        {
+            RemoteTableName remoteTableName = table.getRequiredNamedRelation().getRemoteTableName();
+            return handle.createQuery("""
+                                SELECT
+                                    column_name,
+                                    -- TODO min_value, max_value,
+                                    nulls_ratio
+                                FROM mysql.column_stats
+                                WHERE db_name = :database AND TABLE_NAME = :table_name
+                                AND nulls_ratio IS NOT NULL
+                            """)
+                    .bind("database", remoteTableName.getCatalogName().orElse(null))
+                    .bind("table_name", remoteTableName.getTableName())
+                    .map((rs, ctx) -> {
+                        String columnName = rs.getString("column_name");
+                        double nullsRatio = rs.getDouble("nulls_ratio");
+                        return entry(columnName, new AnalyzeColumnStatistics(nullsRatio));
+                    })
+                    .collect(toImmutableMap(Entry::getKey, Entry::getValue));
+        }
+
+        Map<String, ColumnIndexStatistics> getColumnIndexStatistics(JdbcTableHandle table)
+        {
+            RemoteTableName remoteTableName = table.getRequiredNamedRelation().getRemoteTableName();
+            return handle.createQuery("""
+                                SELECT
+                                    COLUMN_NAME,
+                                    MAX(NULLABLE) AS NULLABLE,
+                                    MAX(CARDINALITY) AS CARDINALITY
+                                FROM INFORMATION_SCHEMA.STATISTICS
+                                WHERE TABLE_SCHEMA = :schema AND TABLE_NAME = :table_name
+                                AND SEQ_IN_INDEX = 1 -- first column in the index
+                                AND SUB_PART IS NULL -- ignore cases where only a column prefix is indexed
+                                AND CARDINALITY IS NOT NULL -- CARDINALITY might be null (https://stackoverflow.com/a/42242729/65458)
+                                AND CARDINALITY != 0 -- CARDINALITY is initially 0 until analyzed
+                                GROUP BY COLUMN_NAME -- there might be multiple indexes on a column
```

Comment:
do we need to port this to MySQL as well?

Reply:
this was copied from MySQL and slightly adapted. 

i think here `AND CARDINALITY != 0` line was added (based on observation with MariaDB)
i suspect MySQL benefits from `AND CARDINALITY IS NOT NULL` line and maybe it's redundant for mariadb. it is very hard to test this, so i retained old condition and just added a new one

someone would need to observe what mysql is doing to determine whether `AND CARDINALITY != 0 ` would make sense there